ethereum/nft/trades/insert_nftx.sql

```diff
@@ -0,0 +1,290 @@
+    'Sell' as tx_type,
+    contract_address as vault,
+    unnest("nftIds") as nfts,
+    "to" as seller,
+    contract_address as buyer
+    from
+    Swapped_t1
+    ORDER BY time
+),
+Swapped_t2R AS
+(
+    select
+    evt_block_time as time,
+    evt_block_number,
+    evt_tx_hash as tx,
+    evt_index,
+    'Swap' as type,
+    'Buy' as tx_type,
+    contract_address as vault,
+    unnest("redeemedIds") as nfts,
+    contract_address as seller,
+    "to" as buyer
+    from
+    Swapped_t1
+    ORDER BY time
+),
+Txs_t as
+(
+    select * from Minted_t2
+    UNION ALL
+    select * from Redeemed_t2
+    UNION ALL
+    select * from Swapped_t2M
+    UNION ALL
+    select * from Swapped_t2R
+),
+Txs_t2 as
+(
+    select * from Txs_t
+    where time >= start_ts AND time < end_ts
+),
+Txs as
+(
+    select * from Txs_t2 LEFT JOIN vaults on Txs_t2.vault = vaults."vaultAddress"
+),
+
+-- #################################################################
+-- #################################################################
+
+
+-- #################################################################
+-- Parse Data
+-- #################################################################
+rows AS (
+    INSERT INTO nft.trades (
+        block_time,
+        nft_project_name,
+        nft_token_id,
+        erc_standard,
+        platform,
+        platform_version,
+        trade_type,
+        number_of_items,
+        category,
+        evt_type,
+        usd_amount,
+        seller,
+        buyer,
+        original_amount,
+        original_amount_raw,
+        original_currency,
+        original_currency_contract,
+        currency_contract,
+        nft_contract_address,
+        exchange_contract_address,
+        tx_hash,
+        block_number,
+        nft_token_ids_array,
+        senders_array,
+        recipients_array,
+        erc_types_array,
+        nft_contract_addresses_array,
+        erc_values_array,
+        tx_from,
+        tx_to,
+        trace_address,
+        evt_index,
+        trade_id
+    )
+
+    SELECT
+        time AS block_time,
+        name AS nft_project_name,
+        nfts AS nft_token_id,
+        'erc20' AS erc_standard,
+        'NFTX' as platform,
+        1 as platform_version,
+        'Single Item Trade' AS trade_type,
+        1 AS number_of_items,
+        tx_type as category,
+        type as evt_type,
+        tok_price_usd AS usd_amount,
+        seller AS seller,
+        buyer AS buyer,
+        tok_price AS original_amount,
+        tok_price*10^18 AS original_amount_raw,
+        'WETH' AS original_currency,
+        '\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'::bytea AS original_currency_contract,
+        '\xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2'::bytea AS currency_contract,
+        "assetAddress" AS nft_contract_address,
+        "vaultAddress" AS exchange_contract_address,
+        tx AS tx_hash,
+        evt_block_number AS block_number,
+        nfts AS nft_token_ids_array,
```

Comment:
for this column (and all below which end in '_array') -- we'll need to force into this data type of text[] on nft trades table. 
please refer to opensea script as reference:
https://github.com/duneanalytics/abstractions/blob/master/ethereum/nft/trades/insert_opensea.sql

here is the CTE which aggregates by transaction, then uses array_agg() function:
![Screen Shot 2022-05-10 at 5 09 06 PM](https://user-images.githubusercontent.com/102681548/167722485-dc5b3d81-1e9d-408a-8e5f-5498c4b8e35a.png)

Reply:
this is then later joined downstream to grab array value